sam.c

```diff
@@ -1088,6 +1088,73 @@ int sam_hdr_write(htsFile *fp, const bam_hdr_t *h)
     return 0;
 }
 
+int sam_hdr_change_HD(bam_hdr_t *h, const char *key, const char *val, const char *sam_version)
+{
+    char *p, *q, *beg = NULL, *end = NULL, *newtext;
+    if (!h || !key)
+        return -1;
+
+    if (h->l_text > 3) {
+        if (strncmp(h->text, "@HD", 3) == 0) { //@HD line exists
+            if ((p = strchr(h->text, '\n')) == 0) return -1;
+            *p = '\0'; // for strstr call
+
+            char tmp[5];
```

Comment:
Could maybe avoid the str* functions below by assigning:
```C
char tmp[5] = { '\t', key[0], key[0] ? key[1] : '\0', ':', '\0' };
```
Could also be simplified if we ensure that key is two chars long first.